### DISCUSSION TITLE
`R` function to detect output format

### DISCUSSION THREAD

**Discussion by vincentarelbundock** [quarto-dev/quarto-cli#3278] [Q&A] [answered]
Hi,

Is there an `R` function I can call in a Quarto code chunk to detect the output format of the notebook? In Rmarkdown, I might do something like:

```r
rmarkdown::default_output_format(knitr::current_input())$name
```

Use-case: my `modelsummary` package draws regression tables, and it must know the target output format in order to create appropriate HTML, LaTeX, or other output. In Rmarkdown, I can dynamically make this adjustment by calling the code above under the hood, and adapting the behavior of `modelsummary` on the fly. Unfortunately, I was unable to find a similar function in Quarto.

Did I miss something? If not, do you think that this feature could be implemented?

Thanks a lot for your work on this great project!

**Comment by mcanouil** [comment]
First of, `knitr` offers `knitr::is_latex_output()` and `knitr::is_html_output()`.

And since Quarto is using `knitr`, then you can use also use those to check if the format is an HTML or LaTeX based format.

**Comment by vincentarelbundock** [comment]
Thanks, that's useful. But is there a more general solution to also detect markdown, word, and other output formats?

**Comment by mcanouil** [comment, marked as answer]
Again, since Quarto is using `knitr`, if you want the output format passed to Pandoc, then `knitr::pandoc_to()`.

You can also have access to all YAML metadata using `rmarkdown::metadata` list, _e.g._, `rmarkdown::metadata[["format"]]`.

**Comment by vincentarelbundock** [comment]
Yep, that should work. Thanks a lot @mcanouil

**Reply by mcanouil** [reply]
You're welcome.
There are a lot more "hidden" functions in `knitr` 😉
